File: LP3/DAA/1_fibonacci_non_recursive.cpp
#include <iostream>
using namespace std;

int fib(int n, int &steps) {
    int a = 0, b = 1, c;
    if (n == 0) return a;
    for (int i = 2; i <= n; i++) {
        steps++;
        c = a + b;
        a = b;
        b = c;
    }
    steps++;
    return b;
}

int main() {
    int n, steps = 0;
    cin >> n;
    cout << "Fibonacci(" << n << ") = " << fib(n, steps) << endl;
    cout << "Steps: " << steps << endl;
    return 0;
}

File: LP3/DAA/1_fibonacci_non_recursive.py
def fib(n, steps):
    a, b = 0, 1
    if n == 0:
        return a
    for i in range(2, n+1):
        steps[0] += 1
        a, b = b, a + b
    steps[0] += 1
    return b

n = int(input())
steps = [0]
result = fib(n, steps)
print(f"Fibonacci({n}) = {result}")
print(f"Steps: {steps[0]}")